source/slang-capture-replay/replay/slang-decoder.cpp

```diff
@@ -0,0 +1,1974 @@
+            break;
+        case ApiCallId::ICompositeComponentType_linkWithOptions:
+            ICompositeComponentType_linkWithOptions(objectId, parameterBlock);
+            break;
+        }
+        return true;
+    }
+
+    bool SlangDecoder::processITypeConformanceMethods(ApiCallId callId, ObjectID objectId, ParameterBlock const& parameterBlock)
+    {
+        switch(callId)
+        {
+        default:
+            slangCaptureLog(LogLevel::Error, "Unhandled Slang API call: %d\n", callId);
+            return false;
+        case ApiCallId::ITypeConformance_getSession:
+            ITypeConformance_getSession(objectId, parameterBlock);
+            break;
+        case ApiCallId::ITypeConformance_getLayout:
+            ITypeConformance_getLayout(objectId, parameterBlock);
+            break;
+        case ApiCallId::ITypeConformance_getSpecializationParamCount:
+            ITypeConformance_getSpecializationParamCount(objectId, parameterBlock);
+            break;
+        case ApiCallId::ITypeConformance_getEntryPointCode:
+            ITypeConformance_getEntryPointCode(objectId, parameterBlock);
+            break;
+        case ApiCallId::ITypeConformance_getTargetCode:
+            ITypeConformance_getTargetCode(objectId, parameterBlock);
+            break;
+        case ApiCallId::ITypeConformance_getResultAsFileSystem:
+            ITypeConformance_getResultAsFileSystem(objectId, parameterBlock);
+            break;
+        case ApiCallId::ITypeConformance_getEntryPointHash:
+            ITypeConformance_getEntryPointHash(objectId, parameterBlock);
+            break;
+        case ApiCallId::ITypeConformance_specialize:
+            ITypeConformance_specialize(objectId, parameterBlock);
+            break;
+        case ApiCallId::ITypeConformance_link:
+            ITypeConformance_link(objectId, parameterBlock);
+            break;
+        case ApiCallId::ITypeConformance_getEntryPointHostCallable:
+            ITypeConformance_getEntryPointHostCallable(objectId, parameterBlock);
+            break;
+        case ApiCallId::ITypeConformance_renameEntryPoint:
+            ITypeConformance_renameEntryPoint(objectId, parameterBlock);
+            break;
+        case ApiCallId::ITypeConformance_linkWithOptions:
+            ITypeConformance_linkWithOptions(objectId, parameterBlock);
+            break;
+        }
+        return true;
+    }
+
+    bool SlangDecoder::processFunctionCall(FunctionHeader const& header, ParameterBlock const& parameterBlock)
+    {
+        return false;
+    }
+
+
+    bool SlangDecoder::ICreateGlobalSession(ObjectID objectId, ParameterBlock const& parameterBlock)
+    {
+        return false;
+    }
+
+    bool SlangDecoder::IGlobalSession_createSession(ObjectID objectId, ParameterBlock const& parameterBlock)
+    {
+        StructDecoder<slang::SessionDesc> sessionDesc;
+        sessionDesc.decode(parameterBlock.parameterBuffer, parameterBlock.parameterBufferSize);
```

Comment:
```
ld: warning: ignoring duplicate libraries: 'Release/lib/libcore.a'
ld: Undefined symbols:
  SlangCapture::StructDecoder<slang::SessionDesc, void>::decode(unsigned char const*, long long), referenced from:
      SlangCapture::SlangDecoder::IGlobalSession_createSession(unsigned long long, SlangCapture::SlangDecoder::ParameterBlock const&) in slang-decoder.cpp.o
      SlangCapture::SlangDecoder::IGlobalSession_getSessionDescDigest(unsigned long long, SlangCapture::SlangDecoder::ParameterBlock const&) in slang-decoder.cpp.o
```

This is weird issue that I haven't seen in gcc and linux clang-15. It only shows up on apple clang-15.

Reply:
The issue is fixed by moving the template explicit instantiation from header file to cpp file.